src/EditorFeatures/CSharp/QuickInfo/SyntacticQuickInfoProvider.cs

```diff
@@ -59,12 +60,20 @@ internal class SyntacticQuickInfoProvider : AbstractQuickInfoProvider
                 return null;
             }
 
-            // If the open brace is the first token of the node (like in the case of a block node or
-            // an accessor list node), then walk up one higher so we can show more useful context
-            // (like the method a block belongs to).
-            if (parent.GetFirstToken() == openBrace)
+            var spanStart = parent.SpanStart;
+            var spanEnd = openBrace.Span.End;
+
+            // If the parent is a scope block, check and include nearby comments around the open brace
```

Comment:
@amcasey I assume scripting will support floating blocks? The current check as proposed only applies to blocks within blocks (the usual case, since regular C# doesn't support blocks floating in nowhere-land), but we might want this logic for standalone blocks in scripting... 

I'm not sure if the previous implementation attempted to do the "right" thing for scripts anyway, but I just wanted to bring it up for discussion.